lib/index.d.ts

```diff
@@ -2,474 +2,596 @@
+  getJointList(): JointEdge | null;
+  /**
+   * Warning: this list changes during the time step and you may miss some
+   * collisions if you don't use ContactListener.
+   */
+  getContactList(): ContactEdge | null;
+  isStatic(): boolean;
+  isDynamic(): boolean;
+  isKinematic(): boolean;
+  /**
+   * This will alter the mass and velocity.
+   */
+  setStatic(): Body;
+  setDynamic(): Body;
+  setKinematic(): Body;
+  /**
+   * @private
+   */
+  getType(): BodyType;
+  /**
+   * @private
+   */
+  setType(type: BodyType): void;
+  isBullet(): boolean;
+  setBullet(flag: boolean): void;
+  isSleepingAllowed(): boolean;
+  setSleepingAllowed(flag: boolean): void;
+  isAwake(): boolean;
+  setAwake(flag: boolean): void;
+  isActive(): boolean;
+  setActive(flag: boolean): void;
+  isFixedRotation(): boolean;
+  setFixedRotation(flag: boolean): void;
+  getTransform(): Transform;
+  setTransform(position: Vec2, angle: number): void;
+  synchronizeTransform(): void;
+  synchronizeFixtures(): void;
+  advance(alpha: number): void;
+  getPosition(): Vec2;
+  setPosition(p: Vec2): void;
+  getAngle(): number;
+  setAngle(angle: number): void;
+  getWorldCenter(): Vec2;
+  getLocalCenter(): Vec2;
+  getLinearVelocity(): Vec2;
+  getLinearVelocityFromWorldPoint(worldPoint: Vec2): Vec2;
+  getLinearVelocityFromLocalPoint(localPoint: Vec2): Vec2;
+  setLinearVelocity(v: Vec2): void;
+  getAngularVelocity(): number;
+  setAngularVelocity(w: number): void;
+  getLinearDamping(): number;
+  setLinearDamping(linearDamping: number): void;
+  getAngularDamping(): number;
+  setAngularDamping(angularDamping: number): void;
+  getGravityScale(): number;
+  setGravityScale(scale: number): void;
+  getMass(): number;
+  getInertia(): number;
+  getMassData(data: MassData): void;
+  resetMassData(): void;
+  setMassData(massData: MassData): void;
+  applyForce(force: Vec2, point: Vec2, wake?: boolean): void;
+  applyForceToCenter(force: Vec2, wake?: boolean): void;
+  applyTorque(torque: number, wake?: boolean): void;
+  applyLinearImpulse(impulse: Vec2, point: Vec2, wake?: boolean): void;
+  applyAngularImpulse(impulse: number, wake?: boolean): void;
+  shouldCollide(that: Body): boolean;
+  createFixture(def: FixtureDef): Fixture;
+  createFixture(shape: Shape, opt?: FixtureOpt): Fixture;
+  createFixture(shape: Shape, density?: number): Fixture;
+  destroyFixture(fixture: Fixture): void;
+  getWorldPoint(localPoint: Vec2): Vec2;
+  getWorldVector(localVector: Vec2): Vec2;
+  getLocalPoint(worldPoint: Vec2): Vec2;
+  getLocalVector(worldVector: Vec2): Vec2;
+}
+
+export type BodyDef = Partial<{
+  type: BodyType,
+  position: Vec2,
+  angle: number,
+  linearVelocity: Vec2,
+  angularVelocity: number,
+  linearDamping: number,
+  angularDamping: number,
+  fixedRotation: boolean,
+  bullet: boolean,
+  gravityScale: number,
+  allowSleep: boolean,
+  awake: boolean,
+  active: boolean,
+  userData: any,
+}>;
+
+export class Body {
```

Comment:
do we need both this and `export interface Body`? it seems this class is a copy of the interface.

Reply:
Good catch! I was experimenting with using classes and committed it by accident.